proto/catalog.proto

```diff
@@ -476,10 +476,13 @@ message Table {
   // in the streaming plan, it only applies to append-only tables.
   optional uint32 retention_seconds = 37;
 
-  // This field specifies the index of the column set in the "with version
+  reserved 38;
+  // deprecated, use "version_column_indices" instead.
+  reserved "version_column_index";
+  // This field specifies the indices of the columns set in the "with version
   // column" within all the columns. It is used for filtering during "on
-  // conflict" operations.
-  optional uint32 version_column_index = 38;
+  // conflict" operations. Multiple columns will be compared lexicographically.
+  repeated uint32 version_column_indices = 47;
```

Comment:
I think `repeated` with `[packed=false]` can be backward compatible with singular field. So there's no need to create a new field here.

- https://protobuf.dev/programming-guides/encoding/#packed
- https://protobuf.dev/programming-guides/proto3/#conditionally-safe-changes

Reply:
Also, it might be noted that there are still table catalog stored in the plan node definition. SQL migration on `Table` might not be sufficient.